### ISSUE TITLE
Mac VO is difficult to use and keeps saying "busy"

### ISSUE THREAD

**Issue by KatieWoe** [phetsims/molecules-and-light#349]
**Device**
Mac Pro
**OS**
Mac 10.15.5
**Browser**
Safari 13.1.1
**Problem Description**
For https://github.com/phetsims/QA/issues/503. Found with Mac VO, does not seem to happen with other sims using Mac VO (checked published version of Friction) which indicates an issue with Molecules and Light, rather than VO itself.
When starting to interact with the sim with Mac VO, depending on how you start the sim may say nothing but "Safari is busy" and "busy." You can interact with the sim, using keyboard nav, but it will say "busy" in place of the descriptions it should give. If you start interaction upon opening the sim by pressing tab and looking that way, things seem ok. However, if you start interaction with ctrl+opt+shift+down and/or arrows to start reading the scene description, this bug occurs. I see this both when starting VO before and after a load of the sim.

**Steps**
1. Go to the sim
2. Start VO
3. Follow VO instructions (ctrl+opt+shift+down) to enter sim
4. Press arrows to attempt to read through description

**Comment by KatieWoe** [comment]
https://drive.google.com/file/d/1ctYMVSDFXtzJe5YUk7p1wDQZgw-fiNHd/view?usp=sharing
On mobile VO, trying a two finger swipe (which should read from the top of the page) instead reads a lot of links and buttons that I didn't really recognize. This continues on for a while (video only covers about half of it). I'm wondering if it has the same cause. 
@jessegreenberg if you think it is separate, let me know and I can make a separate issue.

**Comment by jessegreenberg** [comment]
Interesting, thanks @KatieWoe - yes it may be related but I am not sure. Its like VoiceOver is reading through browsing history or previous webpages or something.

**Comment by terracoda** [comment]
This does not happen on my iPhone7.

**Comment by jessegreenberg** [comment]
I tried for about 10 minutes to get this happen but I have not been able to unfortunately. I still suspect that this is a performance problem.

@terracode @KatieWoe could you please add any computer specific details? Here are mine:
macOS Catalina 10.15.4
MacBook Pro (Late 2013)
Processor 2.4 GHz Dual-Core Intel Core i5
Memory 8 GB 1600 MHz DDR3

What is your battery percentage at when you see this happen? Any power saving settings?

**Comment by jessegreenberg** [comment]
> On mobile VO, trying a two finger swipe

I think this is a different  ssue, Ill make a new one.

**Comment by KatieWoe** [comment]
macOS Catalina 10.15.5 (may have updated since the issue was made, so may have been 10.15.4)
PacBook Pro 2017
Processor 2.3 GHz Dual-Core Intel Core i5
Memory 8 GB 2133 MHz LPDDR3

It would have been between 90 and 100 percent and plugged in.

**Comment by jessegreenberg** [comment]
OK thanks @KatieWoe - pretty similar specs, nothing to indicate your computer is a slower model. Would you mind quickly checking again and seeing if this happens with consistency?

**Comment by KatieWoe** [comment]
It is still happening. In fact, I seemed to be having even more trouble getting it to read anything.

**Comment by jessegreenberg** [comment]
Would you mind trying with this version? (zoom disabled)
https://phet-dev.colorado.edu/html/molecules-and-light/1.5.0-dev.50/phet/molecules-and-light_all_phet.html

**Comment by KatieWoe** [comment]
It does seem to be working with that link when I did a quick check

**Comment by jessegreenberg** [comment]
Trying to get this to happen again, I couldn't get it to say "busy" during normal use. But it would say "busy" if the sim was zoomed in and panned to a new element on change of focus. But after about 2 seconds it was catch up and read the newly focused item.

**Comment by KatieWoe** [comment]
What about when you are trying to listen the scene description? That seemed to be the bigger cause of problems when testing.

**Comment by jessegreenberg** [comment]
Is anything else going on in the sim, or just right after sim load?

**Comment by KatieWoe** [comment]
right after sim load, but it depends on what you do at the beginning. As said in original issue:
>If you start interaction upon opening the sim by pressing tab and looking that way, things seem ok. However, if you start interaction with ctrl+opt+shift+down and/or arrows to start reading the scene description, this bug occurs.

**Comment by jessegreenberg** [comment]
OK, thanks. I still am not able to get this to happen. I tried moving from the navigation bar and reading through the sim with arrow keys, and also reading through everything with "control" + "option" + "A" and I haven't heard it say "busy" yet.

**Comment by jessegreenberg** [comment]
> Press arrows to attempt to read through description

Ahh, I got it to happen by just pressing the arrow keys! The safari command to read content in VoiceOver is to press "control" + "option" + "arrow keys".

But when I just use arrow keys I do hear busy. I am not sure yet what the arrow keys are requesting VO to do or if our sim is doing performance intensive work in response to the arrow keys. Ill investigate.

**Comment by terracoda** [comment]
I thought the command to read with cursor was:
- Control Option Right Arrow?

To interact with an interactive item:
- Control Option Shift Down Arrow... and VO says:
-" Interact with the title of switch"

Hmm, I wonder where, "the title of" is coming from - more VO-MAC silliness. 
Why doesn't VO just say, "Interact with Infrared Light Source switch"? 
VO knows the name of the button and knows it is a switch.

It's not saying "Busy" with Zoom disabled using the link you provided @jessegreenberg.

**Comment by jessegreenberg** [comment]
> It's not saying "Busy" with Zoom disabled using the link you provided @jessegreenberg.

OK, thanks @terracoda I confirmed that, this is helpful.

**Comment by terracoda** [comment]
Yep, with zoom enabled, I get "Safari busy" a lot. I also get am sim content in between the busies. And their seems to be a bit of delay.

Sounds like it is similar for you @jessegreenberg.

**Comment by terracoda** [comment]
OK, now my fan is going crazy. The sim in zoom and VO  is off.

**Comment by jessegreenberg** [comment]
@terracoda what are your thoughts on whether zoom should be supported with VoiceOver? Is that a combination we need to support? It seems like the zoom operations really strain VoiceOver and I would like to know how far to go in working on optimizations.

**Comment by terracoda** [comment]
@jessegreenberg, that's a good question. Zoom is likely most useful for people with Low Vision who may or may not be using screen reader software.

Screen reader software is most often used by people who blind who who do not have enough vision to read text on the screen.

I think we can focus on making sure the sim works well with VO on when not Zoomed.

I hear "busy" and often lose access to Context Responses. 
I thought the loss of Context Responses was just an Apple issue which we have seen before, but it seems to happen more frequently now. It could be related to the Zoom feature, too?

Let's get VO working with well whne learner is not zoomed in. 

@KatieWoe, do you get "busy" with VO on iOS? Does VO do the same thing if you use zoom on iOS? 

If VO & Zoom work better on iOS at least we have a back-up option.

**Comment by KatieWoe** [comment]
I don't think I saw it in iOS. I never tried using a screen reader while zoomed in, it just did it if it was possible to zoom in.

**Comment by jessegreenberg** [comment]
The above commit stops VoiceOver from saying "busy" when at max zoom and pressing the arrow keys.

**Comment by jessegreenberg** [comment]
> https://github.com/phetsims/molecules-and-light/issues/349#issuecomment-644853826

OK, that sounds good to me @terracoda, thanks.

**Comment by jessegreenberg** [comment]
Here is a new version with this fix:

https://phet-dev.colorado.edu/html/molecules-and-light/1.5.0-dev.53/phet/molecules-and-light_en_phet.html?zoom=true

@KatieWoe and @terracoda can you test to see if it is improved in this version?

**Comment by terracoda** [comment]
@jessegreenberg, I think you are amazing.
I played with sim without zooming and I never heard "busy" once.
I then zoomed, and I heard "busy", but I think I hard it less than before.

As discussed in our meeting today, I'll reach out to @emily-phet. And I as we discussed we will have more control and learners will have more options when they combine a Self-Voicing option with the Zoom option.

I'll note here, too, that you mentioned that on iOS, the zoom feature works well, though one has to Zoom first and then enable VO. One cannot Zoom after VO is enabled.

**Comment by emily-phet** [comment]
@terracoda and @jessegreenberg I think it's fine to proceed with RC with less than ideal experience combining Zoom and VO.

**Comment by terracoda** [comment]
Ok, @jessegreenberg, I think this issue is ready to close, so closing.
